Remove install file and harden sytemd unit a bit

brunnre8 opened the pull request:
This removes the install file and replaces it with the systemd-sysuser / tmpfiles factories.

Further this hopefully splits the configs from the "data" directories by symlinking it like you suggested in IRC (I am bookworm)

Could you please review and tell me what you'd like to see changed?

Sorry for the delay, took a while due to $life

brunnre8 committed:
Remove install file and split config/data dirs

xPaw commented:
I don't quite like the idea of using symlinks, as that diverses from all the other packaging methods Lounge has, and would only cause more harm than good (considering its symlinked, not actually moved).

xPaw commented on .SRCINFO:
```diff
@@ -13,9 +12,13 @@ pkgbase = thelounge
-	sha256sums = SKIP
+	source = sysusers.d
+	source = tmpfiles.d
+	md5sums = f2be61e721cc677fb624687cf1e8944c
```
Can you revert to using `sha256sums`?

brunnre8 replied:
yes, will do

xPaw commented on PKGBUILD:
```diff
@@ -1,46 +1,48 @@
-	'4b4970caba850042244798008afd334dd2ddd7e25339c864837ce11ba4703021'
-	'SKIP'
-	'SKIP'
+    "http://registry.npmjs.org/$pkgname/-/$pkgname-$pkgver.tgz"
```
https

brunnre8 replied:
sorry, oversight

xPaw commented:
Have you tried installing thelounge@next with this? Does it correctly handle installing/downloading sqlite3?

brunnre8 commented:
@xPaw the split config / data dirs was a suggestion via IRC.

Can you discuss between the maintainers what you prefer?

And no I didn't try next as the current npm release was broken due to a broken dependency (event-stream) when I tried

xPaw commented:
> Can you discuss between the maintainers what you prefer?

Our general preference is to have minimal differences between packages, so just using /etc without symlinks would be preferable. Maybe in a future major release we can change it.

brunnre8 commented:
Fair, will change it.

@xPaw do you plan a new release on npm or is the build failure pebcak on my side?

xPaw commented:
> @xPaw do you plan a new release on npm or is the build failure pebcak on my side?

It's certainly broken via yarn if you try to pull dev dependencies, however I'm not sure why you're running into that with npm, can you try with `--production` flag? You might also need `--unsafe-perm` to get sqlite to work.

brunnre8 commented:
I tried with yarn during testing, run into the issue and then didn't try npm as well.

I'll fix it up according to your suggestions

brunnre8 committed:
Revert split config/data dirs

And address the other review comments

brunnre8 commented:
Addressed the review comments.

@xPaw rc5 builds fine with the pkgbuild via the npm registry, but I run Arch on x86 so that should be fairly straight forward.

brunnre8 committed:
Update to 3.0.0

maxpoulin64 commented:
Looks good to me and verified working! Merging in a moment

Superpiffer commented:
Forse some reason, user.service file has a wrong sha256sum:
```
sha256sum user.service
c609f3309f54bd6285e99ff29ca2464828bec7bbbca67243ee688bd2d605dbf0  user.service
```

brunnre8 commented:
Ack, the aur version doesn't have the updated version

@maxpoulin64  as I don't have push access yet, could you please have a look?
The git repo works fine, just the aur is broken

maxpoulin64 commented:
That's odd, it worked for me and they should both be the same master branch.

maxpoulin64 commented:
Seems to have been because master had #7 merged in so the checksums had changed between what was in this PR and what ended up in master.
